### PR TITLE
feat: Save report runs [PT-188507981]

### PR DESCRIPTION
**Pull request by dougmartin**
This adds a new table in the database that saves report runs and two new liveview pages that list report runs and show a report run.

### CODE REVIEW

File: server/lib/report_server_web/components/custom_components.ex [hunk 1]
```diff
@@ -163,4 +163,94 @@ defmodule ReportServerWeb.CustomComponents do
...
+            <th class="p-2 font-normal border-b">Run</th>
+            <th class="p-2 font-normal border-b" :if={@include_user}>User</th>
+            <th class="p-2 font-normal border-b" :if={@include_report_titles}>Report</th>
+            <th class="p-2 font-normal border-b">Filters</th>
+            <th class="p-2 font-normal border-b">Ran</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr :for={report_run <- @report_runs} class="group hover:bg-zinc-200 even:bg-gray-50">
+            <td class="p-2 font-normal border-b align-top"><.link class="underline" href={"/new-reports/runs/#{report_run.id}"}><%= report_run.id %></.link></td>
```
Comment (bgoldowsky) [inline server/lib/report_server_web/components/custom_components.ex:241]: The ID number is kind of meaningless to the end user. OK to include it for test/debug purposes, but clicking it to re-run the report is not very obvious.  There should probably be a "re-run" button instead.

**Review by bgoldowsky — APPROVED** [review]
This looks great.
I made a couple of suggestions that can be part of a future PR.
Also next time we pair I'd like to figure out why the tests aren't running for me.

File: server/lib/report_server_web/live/new_report_live/index.html.heex [hunk 2]
```diff
@@ -4,12 +4,14 @@
...
   </.header>
 
-  <div>
-    <.described_link navigate={report.path} description={report.subtitle} :for={report <- @report_group.children}>
-      <%= report.title %>
-      <span :if={report.tbd}>🚧🛠️</span>
-    </.described_link>
-  </div>
+  <.described_link navigate={~p"/new-reports/runs"} description={if @user.portal_is_admin, do: "Lists all the report runs (because you are an admin)", else: "Lists your report runs"} :if={@is_root}>
+    <%= if @user.portal_is_admin, do: "All Runs", else: "Your Runs" %>
```
[the remark this answers is not in the document]
Comment (dougmartin) [reply server/lib/report_server_web/live/new_report_live/index.html.heex]: They do - that was in the follow on commit I made this morning.

File: server/lib/report_server_web/components/custom_components.ex [hunk 3]
```diff
@@ -163,4 +163,94 @@ defmodule ReportServerWeb.CustomComponents do
...
+        minutes = div(diff, 60)
+        "#{minutes} #{pluralize(minutes, "minute")} ago"
+
+      diff < 86_400 ->
+        hours = div(diff, 3600)
+        "#{hours} #{pluralize(hours, "hour")} ago"
+
+      true ->
+        days = div(diff, 86_400)
+        "#{days} #{pluralize(days, "day")} ago"
```
[the remark this answers is not in the document]
Comment (dougmartin) [reply server/lib/report_server_web/components/custom_components.ex:209]: Good idea.

File: server/lib/report_server_web/components/custom_components.ex (hunk 1 above)
Comment (dougmartin) [reply server/lib/report_server_web/components/custom_components.ex:241]: I can see Trudi memorizing the report run ids she wants to run during a class test.  We could also add a view buttom on the far right.

**Pull request merged**